core/dataloader/tabdataloader.go

```diff
@@ -0,0 +1,161 @@
+// Copyright 2024 Dolthub, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package dataloader
+
+import (
+	"bufio"
+	"fmt"
+	"io"
+	"strings"
+
+	"github.com/dolthub/go-mysql-server/sql"
+	"github.com/sirupsen/logrus"
+
+	"github.com/dolthub/doltgresql/server/types"
+)
+
+// TabularDataLoader tracks the state of a load data operation from a tabular data source.
+type TabularDataLoader struct {
```

Comment:
There shouldn't be any changes to this code: it was just moved to this new file from `dataloader.go`